### DISCUSSION TITLE
"v-model cannot be used on a prop" error with Vue 3 (>= 3.2.45)

### DISCUSSION THREAD

**Discussion by nulele** [quasarframework/quasar#17112] [General - Components / Directives / etc] [answered]
Hello,

I'd like to upgrade my quasar 2 (2.15.2) project to every package last version but when it comes to upgrade Vue I get an error:

`v-model cannot be used on a prop, because local prop bindings are not writable`

Since quasar documentation encourages the `v-model` approach on components (such as QInput, QDialog...), I'm wondering if it's normal that this is only possibile with Vue 3 version before 3.2.45.

Now, in this project I have hundred of component files with this "problem". Does this means that my project will be stucked to this Vue 3.2.45 version from now on?

**Comment by yusufkandemir** [comment]
I doubt it's related to Quasar. In order to get help, you can share one out of "hundred of component files with this 'problem'"

**Reply by nulele** [reply, marked as answer]
Well, perhaps I was misunderstood, I didn't mean to say it's directly related to quasar.

Anyway, looks like that the whole problem revolved around `props` only.

I have several QDialog triggered by and QInput valued by props variables:

```
<q-dialog position="bottom" persistent v-model="toggleDialog">
    ...
</q-dialog>

<q-input
    filled
    readonly
    :label="$t('label.client')"
    v-model="clientName"
/>

```

```
props: {
    toggleDialog: {
        default: false,
    },
    clientName: {
        required: false,
    }
}
```
It was enough to replace `v-model` with `:modelValue` and the compiling error disappeared.

**Comment by rstoenescu** [comment]
Hi,

It only means that you are incorrectly using the v-model.
We can help if you post a codepen/jsfiddle/stackblitz/repo that we can look at and point you to the right direction.
